packages/typespec-test/test/hierarchy_generic/generated/typespec-ts/src/FooClient.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Pipeline } from "@azure/core-rest-pipeline";
+import { A } from "./models/models.js";
+import { Op1Options } from "./models/options.js";
+import { getBOperations, BOperations } from "./classic/b/index.js";
+import { getDOperations, DOperations } from "./classic/d/index.js";
+import { createFoo, FooClientOptions, FooContext, op1 } from "./api/index.js";
+
+export { FooClientOptions } from "./api/FooContext.js";
+
+export class FooClient {
+  private _client: FooContext;
+  /** The pipeline used by this client to make requests */
+  public readonly pipeline: Pipeline;
+
+  constructor(endpoint: string, options: FooClientOptions = {}) {
+    this._client = createFoo(endpoint, options);
+    this.pipeline = this._client.pipeline;
+    this.b = getBOperations(this._client);
+    this.d = getDOperations(this._client);
+  }
+
+  op1(body: A, options: Op1Options = { requestOptions: {} }): Promise<void> {
+    return op1(this._client, body, options);
+  }
+
+  /** The operation groups for BB */
+  public readonly b: BOperations;
```

Comment:
I think we'll have to define this like

```ts
public readonly b: {
  b: {
    op1: (body: BA, options?: BBOp1Options) => Promise<void>;
  };
}
```

To avoid the need to export the Operations interface which is really junk for our customers

Reply:
As discussed offline, we should export the operation interface for now, otherwise we will have ugly classical client in the docs site. /cc  @bterlson